File: demo/ChengFenStore/Data/PaymentRepository.cs
using ChengFenStore.Models;
using Microsoft.EntityFrameworkCore;
using System.Collections.Generic;
using System.Linq;

namespace ChengFenStore.Data
{
    public interface IPaymentRepository
    {
        void AddPayment(Payment payment);
        List<Payment> GetPayments();
        Payment GetPaymentById(int paymentId);
    }

    public class PaymentRepository : IPaymentRepository
    {
        private readonly AppDbContext _context;

        public PaymentRepository(AppDbContext context)
        {
            _context = context;
        }

        public void AddPayment(Payment payment)
        {
            _context.Payments.Add(payment);
            _context.SaveChanges();
        }

        public List<Payment> GetPayments()
        {
            return _context.Payments.ToList();
        }

        public Payment GetPaymentById(int paymentId)
        {
            return _context.Payments.FirstOrDefault(p => p.PaymentId == paymentId);
        }
    }
}

File: demo/ChengFenStore/Data/PaymentRepository.py
from ..Data.AppDbContext import db, Payment

class PaymentRepository:
    def add_payment(self, payment):
        db.session.add(payment)
        db.session.commit()

    def get_payments(self):
        return Payment.query.all()

    def get_payment_by_id(self, payment_id):
        return Payment.query.get(payment_id)
